### ISSUE TITLE
Import passwords from other password manager automatically

### ISSUE THREAD

**Issue by FrozenAssassine** [FrozenAssassine/EasePass#55]
List of password manager to do:

- [ ] Firefox

**Comment by Flamifly** [comment]
New Plugin for Importing CSV Files
You can define which Properties of the PasswordItem should be filled if you specify a Header with the names of them
If you do not the default design is DisplayName, UserName, EMail, Password, Website, Notes

Because we need to Import a File were should the default Destination to look for Files?

https://github.com/Flamifly/EasePassCsvPasswordImporter

**Comment by Flamifly** [comment]
My Solution probably will be:

Create a new Folder "C:\Users\user\Documents\EasePass\CSV"
And create an Example CSV.
Every CSV File inside the Folder will be imported

**Comment by Flamifly** [comment]
The Plugin will change I will add CSV and XML support

**Comment by FrozenAssassine** [comment]
We have to wait for @finn-freitag to answer your questions, since he implemented all this stuff

**Comment by Flamifly** [comment]
And when will the Plugin be executed btw?
So where will the Import method will be called in EasePass?

**Comment by finn-freitag** [comment]
Maybe you can call an OpenFileDialog to get the path for the CSV file.

The functionality of your plugin is called when you have added your plugin in the plugin management page of the settings and then click on "Import Passwords" in the settings page.

![grafik](https://github.com/user-attachments/assets/29deca41-ec47-4d7e-8a8b-f3a58762283c)

**Comment by Flamifly** [comment]
You mean calling it inside the Plugin?

**Comment by finn-freitag** [comment]
Yes, you need to call the OpenFileDialog within your plugin. Sometimes it is a bit difficult as some .NET dependencies cannot be found in the plugins, but try it.

**Comment by Flamifly** [comment]
Yea I will try it thanks for the Advice
Is there any other File Format which should be supported?

**Comment by finn-freitag** [comment]
I can't think of any other formats at the moment. A while ago, I've tried to create a *.kbdx file importer to import passwords from Keepass but that seems a bit tricky.

**Comment by Flamifly** [comment]
Is it even possible without knowing the Master Password to Import a kbdx file?

I saw that Keepass Supports exports in XML and CSV so maybe that could allow importing Passwords from Keepass need to take a Look on that

**Comment by finn-freitag** [comment]
It is of course not possible to get passwords from a kbdx file without the password. To import, the user would have to enter the master password.

**Comment by Flamifly** [comment]
Should I allow XML/CSV Files from KeePass in the Plugin I Do at the Moment or Should it be it's own Plugin?

**Comment by finn-freitag** [comment]
I would say you can do that in the same Plugin because it's that same file format.

**Comment by Flamifly** [comment]
Finished my Plugin
https://github.com/Flamifly/EasePassFilePasswordImporter

Did not test the JSON and XML Part because it's like 3 lines of code and I think that can't mess up that much

You can now Import CSV, XML and JSON Files
At the Moment only EasePass PasswordItems are allowed (This will be used for Serialize and Desirialize the XML and JSON File)

I will add support for Keepass later

**Comment by Flamifly** [comment]
Throws an Error if I want to execute it from EasePass do you know why @finn-freitag?

**Comment by finn-freitag** [comment]
It's this dependency issue I've told you earlier. The OpenFileDialog you use is part of System.Windows.Forms. This library is not available to plugins (I don't really know why). I will try to find a solution for this later or in the next days.
![grafik](https://github.com/user-attachments/assets/8e9a7942-6bc4-4462-bb39-f06b56ed35c7)

**Comment by Flamifly** [comment]
Thanks for the Respond.
If EasePass would use WinForms could it work then?

**Comment by finn-freitag** [comment]
I'm not sure

**Comment by finn-freitag** [comment]
@Flamifly I added a new interface for plugins to use file pickers. Please use the latest build of "EasePassExtensibility.dll" in you plugin and change your code like this:

![grafik](https://github.com/user-attachments/assets/75c9ed25-bae4-451d-b7f9-aa10b1afbad7)

Please test your plugin intensively. After that, please let me know and I will add your plugin to the Ease Pass Plugin Store.

**Comment by Flamifly** [comment]
Thanks the FilePicker works now 👍🏻

**Comment by Flamifly** [comment]
XML and JSON Import work (kinda slow compared to CSV)

The CSV Import also seems to work fine to me.
The only thing that can cause errors is that the Plugin does not check if the values are valid.
For example for the TOTP if it is base64 or other things but I would say it's up to the User not the Plugin

(Imported for each Plugin 25 Passwords and once with the CSV Import 1000)

**Comment by Flamifly** [comment]
CSV Import:
Tested it with Header and no Header both worked

**Comment by finn-freitag** [comment]
Your plugin should now appear in the Ease Pass Plugin Store.
